DatabaseSystem.Ordered/OrderedRecordManager.cpp

```diff
@@ -0,0 +1,871 @@
+                }
+
+                if (Column::Equals(column, value, data))
+                {
+                    enteredRange = true; // found something equal to data, we are inside the range
+                    auto newRecord = new Record(schema);
+                    memcpy(newRecord->GetData()->data(), currentRecord->GetData()->data(), schema->GetSize());
+                    records.push_back(newRecord);
+                }
+            }
+        }
+        // at this point, the data is not present in the main file OR we found all records equal to data in the main file OR we stopped somewhere in the extension file
+        // there might still be records in the range in the extension file
+
+        auto mainFileblocksCount = m_File->GetHead()->GetBlocksCount();
+        if (m_NextReadBlockNumber < mainFileblocksCount || currentRecord == nullptr) // we stopped somewhere in the main file
+        {
+            // we found all records equal to data in the main file
+            // move to the star of extension file and continue
+            MoveToExtension();
+            currentRecord = new Record(schema);
+        }
+
+
+        // m_NextBlockNumber is somewhere in the extension file
+        // linear search extension file continuing from that point
+        while (MoveNext(currentRecord, accessedBlocks, blockId, recordNumberInBlock))
+        {
+            m_LastQueryBlockAccessCount += accessedBlocks;
+            auto value = schema->GetValue(currentRecord->GetData(), columnId);
+
+            if (Column::Equals(column, value, data))
+            {
+                auto newRecord = new Record(schema);
+                memcpy(newRecord->GetData()->data(), currentRecord->GetData()->data(), schema->GetSize());
+                records.push_back(newRecord);
+            }
+        }
+        return records;
+    }
+    // if the file is not ordered by id, linear search everything
+    return BaseRecordManager::SelectWhereEquals(columnId, data);
+}
+
+void OrderedRecordManager::Delete(unsigned long long id)
+{
+    auto schema = GetSchema();
+    auto recordSize = schema->GetSize();
+    if (m_DeletedRecords * recordSize == m_MaxPercentEmptySpace * GetSize())
+    {
+        Compress();
+    }
+    auto record = Select(id);
+    auto blockId = m_NextReadBlockNumber - 1;
+    auto recordNumberInBlock = m_ReadBlock->GetPosition() - 1;
+
+    if (m_OrderedByColumnId == 0 && 
+        m_NextReadBlockNumber < m_File->GetHead()->GetBlocksCount()) // did a binary search and found the record in main file
+    {
+        blockId = m_NextReadBlockNumber;
+        recordNumberInBlock = m_ReadBlock->GetPosition() - 1;
+    }
+    DeleteInternal(blockId, recordNumberInBlock);
+}
+
+int OrderedRecordManager::DeleteWhereEquals(unsigned int columnId, span<unsigned char> data)
+{
+    auto schema = GetSchema();
+    auto recordSize = schema->GetSize();
+    if (m_DeletedRecords * recordSize == m_MaxPercentEmptySpace * GetSize())
```

Comment:
Não deveria ser >= ?
```suggestion
    if (m_DeletedRecords * recordSize >= m_MaxPercentEmptySpace * GetSize())
```